Sources/Orbit/Support/TextConcatenation/DynamicTextWrapper.swift

```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+
+struct DynamicTextWrapper: View {
+    @Environment(\.sizeCategory) var sizeCategory
+
+    let content: (ContentSizeCategory) -> SwiftUI.Text?
+
+    var body: some View {
+        if let content = content(sizeCategory) {
+            content
+        }
+    }
+}
+
+extension DynamicTextWrapper: SwiftUITextRepresentable {
+    func asText(configuration: ContentSizeCategory) -> SwiftUI.Text? {
+        content(configuration)
+    }
+
+    var wrapped: DynamicTextWrapper {
+        self
+    }
+}
+
+func +(
+    left: DynamicTextWrapper,
+    right: DynamicTextWrapper
+) -> DynamicTextWrapper {
```

Comment:
The declaration comfortably fits on a single line.

Reply:
done